### PR TITLE
TSK-977: Add linting to build pipeline

### PR DESCRIPTION
**Pull request by BVier**

### CODE REVIEW

File: web/src/app/administration/workbasket/details/access-items/access-items.component.spec.ts [hunk 1]
```diff
@@ -42,8 +42,11 @@ describe('AccessItemsComponent', () => {
     configureTests(configure).then(testBed => {
       fixture = TestBed.createComponent(AccessItemsComponent);
       component = fixture.componentInstance;
-      component.workbasket = new Workbasket('1', '', '', '', ICONTYPES.TOPIC, '', '', '', '', '', '', '', '', '', '', '', '',
-        new Links(undefined, undefined, { 'href': 'someurl' }));
+      component.workbasket = new Workbasket('1');
+      component.workbasket.type = ICONTYPES.TOPIC;
+      component.workbasket._links = new Links()
+      component.workbasket._links.accessItems = { 'href': 'someurl' };
```
Comment (benjamineckstein) [inline web/src/app/administration/workbasket/details/access-items/access-items.component.spec.ts]: I like that much more ... good catch ;)

File: web/src/app/administration/workbasket/details/distribution-targets/distribution-targets.component.ts [hunk 2]
```diff
@@ -140,16 +140,16 @@ export class DistributionTargetsComponent implements OnChanges, OnDestroy {
...
 
   performFilter(dualListFilter: any) {
-    dualListFilter.side === Side.RIGHT ? this.distributionTargetsRight = undefined : this.distributionTargetsLeft = undefined;
-    this.onRequest(dualListFilter.side, false);
-    this.workbasketFilterSubscription = this.workbasketService.getWorkBasketsSummary(true, undefined, undefined, undefined,
-      dualListFilter.filterBy.filterParams.name, dualListFilter.filterBy.filterParams.description, undefined,
-      dualListFilter.filterBy.filterParams.owner,	dualListFilter.filterBy.filterParams.type, undefined,
-      dualListFilter.filterBy.filterParams.key, undefined, true).subscribe(resultList => {
+    dualListFilter.side === Side.RIGHT ? delete this.distributionTargetsRight : delete this.distributionTargetsLeft;
+    this.onRequest(false, dualListFilter.side);
```
Comment (benjamineckstein) [inline web/src/app/administration/workbasket/details/distribution-targets/distribution-targets.component.ts]: is this changed on purpose?

File: web/src/app/administration/workbasket/details/distribution-targets/distribution-targets.component.ts [hunk 3]
```diff
@@ -140,16 +140,16 @@ export class DistributionTargetsComponent implements OnChanges, OnDestroy {
...
+    this.workbasketFilterSubscription = this.workbasketService.getWorkBasketsSummary(true, '', '', '',
+      dualListFilter.filterBy.filterParams.name, dualListFilter.filterBy.filterParams.description, '',
+      dualListFilter.filterBy.filterParams.owner,	dualListFilter.filterBy.filterParams.type, '',
+      dualListFilter.filterBy.filterParams.key, '', true).subscribe(resultList => {
         (dualListFilter.side === Side.RIGHT) ?
           this.distributionTargetsRight = (resultList.workbaskets) :
           this.distributionTargetsLeft = (resultList.workbaskets);
-        this.onRequest(dualListFilter.side, true);
+        this.onRequest(true, dualListFilter.side);
```
Comment (benjamineckstein) [inline web/src/app/administration/workbasket/details/distribution-targets/distribution-targets.component.ts]: guess so.

File: web/src/app/administration/classification/details/classification-details.component.spec.ts [hunk 4]
```diff
@@ -67,9 +67,8 @@ describe('ClassificationDetailsComponent', () => {
       spyOn(classificationCategoriesService, 'getCategories').and.returnValue(of(['firstCategory', 'secondCategory']));
       spyOn(classificationsService, 'deleteClassification').and.returnValue(of(true));
       spyOn(classificationCategoriesService, 'getCategoryIcon').and.returnValue(new Pair('assets/icons/categories/external.svg'));
-      component.classification = new ClassificationDefinition('id1', undefined, undefined, undefined, undefined, undefined, undefined,
-        undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined,
-        undefined, undefined, undefined, undefined, new LinksClassification({ 'self': '' }));
+      component.classification = new ClassificationDefinition('id1');
```
Comment (benjamineckstein) [inline web/src/app/administration/classification/details/classification-details.component.spec.ts]: much better ;)

File: web/src/app/administration/access-items-management/access-items-management.component.ts [hunk 5]
```diff
@@ -155,6 +155,14 @@ export class AccessItemsManagementComponent implements OnInit, OnDestroy {
     );
   }
 
+  private addIfExists(item?: string) {
```
Comment (benjamineckstein) [inline web/src/app/administration/access-items-management/access-items-management.component.ts]: is there a better way? don't like meaningless catches

File: web/src/app/administration/workbasket/details/information/workbasket-information.component.spec.ts [hunk 6]
```diff
@@ -138,7 +138,7 @@ describe('WorkbasketInformationComponent', () => {
   }));
 
   it('should post a new workbasket when no workbasketId is defined and update workbasket', async(() => {
-    const workbasket = new Workbasket(undefined, 'created', 'keyModified', 'domain', ICONTYPES.TOPIC, 'modified', 'name', 'description',
+    const workbasket = new Workbasket('', 'created', 'keyModified', 'domain', ICONTYPES.TOPIC, 'modified', 'name', 'description',
```
Comment (benjamineckstein) [inline web/src/app/administration/workbasket/details/information/workbasket-information.component.spec.ts]: is undefined the same as an empty string? what about null?

File: web/src/app/administration/workbasket/details/workbasket-details.component.ts [hunk 7]
```diff
@@ -46,31 +46,31 @@ export class WorkbasketDetailsComponent implements OnInit, OnDestroy {
...
     this.workbasketSelectedSubscription = this.service.getSelectedWorkBasket().subscribe(workbasketIdSelected => {
-      this.workbasket = undefined;
+      delete this.workbasket;
       this.getWorkbasketInformation(workbasketIdSelected);
     });
 
     this.routeSubscription = this.route.params.subscribe(params => {
       let id = params['id'];
-      this.action = undefined;
+      delete this.action;
```
Comment (benjamineckstein) [inline web/src/app/administration/workbasket/details/workbasket-details.component.ts]: what about a first if(!id) return ... 

would be easier on all other ifs.

File: web/src/app/administration/workbasket/master/workbasket-list-toolbar/workbasket-list-toolbar.component.ts [hunk 8]
```diff
@@ -49,7 +49,7 @@ export class WorkbasketListToolbarComponent implements OnInit {
   }
 
   addWorkbasket() {
-    this.workbasketService.selectWorkBasket(undefined);
+    this.workbasketService.selectWorkBasket();
```
Comment (benjamineckstein) [inline web/src/app/administration/workbasket/master/workbasket-list-toolbar/workbasket-list-toolbar.component.ts]: so much better ;)

File: web/src/app/models/page.ts [hunk 9]
```diff
@@ -1,8 +1,8 @@
 export class Page {
     constructor(
-        public size: number = undefined,
-        public totalElements: number = undefined,
-        public totalPages: number = undefined,
-        public number: number = undefined
+        public size: number = 0,
+        public totalElements: number = 0,
+        public totalPages: number = 0,
+        public number: number = 0
```
Comment (benjamineckstein) [inline web/src/app/models/page.ts]: should that have default values at all?

File: web/src/app/workplace/models/task.ts [hunk 10]
```diff
@@ -4,44 +4,44 @@ import {ObjectReference} from './object-reference';
...
+              public classificationSummaryResource: Classification = null,
+              public businessProcessId: string = '',
+              public parentBusinessProcessId: string = '',
+              public owner: string = '',
+              public created: string = '',    // ISO-8601
+              public claimed: string = '',    // ISO-8601
+              public completed: string = '',  // ISO-8601
+              public modified: string = '',   // ISO-8601
+              public planned: string = '',    // ISO-8601
+              public due: string = '',        // ISO-8601
```
Comment (benjamineckstein) [inline web/src/app/workplace/models/task.ts]: is this intended as empty string and not null?

File: web/src/app/administration/access-items-management/access-items-management.component.ts (hunk 5 above)
Comment (BVier) [reply web/src/app/administration/access-items-management/access-items-management.component.ts]: thanks for asking - I forgot to remove this changes!

File: web/src/app/administration/workbasket/details/distribution-targets/distribution-targets.component.ts (hunk 2 above)
Comment (BVier) [reply web/src/app/administration/workbasket/details/distribution-targets/distribution-targets.component.ts]: yes

File: web/src/app/administration/workbasket/details/information/workbasket-information.component.spec.ts (hunk 6 above)
Comment (BVier) [reply web/src/app/administration/workbasket/details/information/workbasket-information.component.spec.ts]: It's not the same but undefined is better. So I took this changes back

File: web/src/app/administration/workbasket/master/workbasket-list-toolbar/workbasket-list-toolbar.component.ts (hunk 8 above)
Comment (BVier) [reply web/src/app/administration/workbasket/master/workbasket-list-toolbar/workbasket-list-toolbar.component.ts]: Thanks! :)

File: web/src/app/workplace/models/task.ts (hunk 10 above)
Comment (BVier) [reply web/src/app/workplace/models/task.ts]: I corrected it

**Comment by BVier** [comment]
@benjamineckstein could you review it again?
I think I solved the recommended problems...

File: web/src/app/administration/workbasket/details/workbasket-details.component.ts (hunk 7 above)
Comment (mustaphazorgati) [reply web/src/app/administration/workbasket/details/workbasket-details.component.ts]: @BVier why did you resolve this?

**Pull request merged**